pilot/cmd/pilot-discovery/main.go

```diff
@@ -103,6 +103,16 @@ func init() {
 		"Controller resync interval")
 	discoveryCmd.PersistentFlags().StringVar(&serverArgs.Config.ControllerOptions.DomainSuffix, "domain", "cluster.local",
 		"DNS domain suffix")
+
+	defaultID := "<unknown>"
+	podName := os.Getenv("POD_NAME")
+	podNamespace := os.Getenv("POD_NAMESPACE")
+	if podName != "" && podNamespace != "" {
+		defaultID = fmt.Sprintf("k8s://%s.%s", podName, podNamespace)
```

Comment:
nit: should we just type out `kubernetes`? iirc, uids that are sent to Mixer use a prefix of `kubernetes://`